### PR TITLE
Add scripts for e2e test on psi 

### LINKED ISSUE #4408: Verify automation process to run macOS tests on PSI
/kind user-story

## User Story
As a QE I want to analyse/research and automate macOS test on PSI.

## Acceptance Criteria 
- [ ] Mention the steps needed for automation
- [ ] Automate the process of test

## Links 
- Feature Request: N/A
- Related issue: #2540 

/kind user-story

### LINKED ISSUE #4412: Automate integration and unit test for windows on PSI
/kind user-story

## User Story
As a QE I want to automate both integration and unit test on Windows.

## Acceptance Criteria 
- [ ] Automate the test for Windows.
- [ ] Resolve the blockers, if any.

## Links 
- Feature Request: N/A
- Related issue: https://github.com/openshift/odo/issues/4379 , #2540

/kind user-story

### PR DESCRIPTION
**Pull request by anandrkskd**
Signed-off-by: anandrkskd <anandrkskd@gmail.com>

**What type of PR is this?**
/kind feature

**What does this PR do / why we need it**:
This PR adds scripts required for automating windows and mac tests for PR

**Which issue(s) this PR fixes**:

Fixes - part of #4412
Fixes - part of #4408 

**PR acceptance criteria**:

- [ ] Unit test 

- [ ] Integration test 

- [ ] Documentation 

- [x] I have read the [test guidelines](https://github.com/openshift/odo/blob/master/docs/dev/test-architecture.adoc)

**How to test changes / Special notes to the reviewer**:
N/A

### CODE REVIEW

**Comment by anandrkskd** [comment]
We are facing failures on PSI tests due to some known issues 
Issue 1. https://github.com/openshift/odo/issues/2877
```
[ssh:Windows 10] [2] + Failure [244.628 seconds]
[ssh:Windows 10] [2] odo java e2e tests
[ssh:Windows 10] [2] C:/Users/Admin/amqp_ci_rcv_odo-windows-e2e-pr-build_PR_4658/repo/tests/e2escenarios/e2e_java_test.go:11
[ssh:Windows 10] [2]   odo component creation
[ssh:Windows 10] [2]   C:/Users/Admin/amqp_ci_rcv_odo-windows-e2e-pr-build_PR_4658/repo/tests/e2escenarios/e2e_java_test.go:59
[ssh:Windows 10] [2]     Should be able to deploy a spring boot uberjar file using openjdk [It]
[ssh:Windows 10] [2]     C:/Users/Admin/amqp_ci_rcv_odo-windows-e2e-pr-build_PR_4658/repo/tests/e2escenarios/e2e_java_test.go:96
[ssh:Windows 10] [2] 
[ssh:Windows 10] [2]     No future change is possible.  Bailing out early after 240.681s.
[ssh:Windows 10] [2]     Running odo.exe with args [odo push --context C:\Users\Admin\849480791]
[ssh:Windows 10] [2]     Expected
[ssh:Windows 10] [2]         <int>: 1
[ssh:Windows 10] [2]     to match exit code:
[ssh:Windows 10] [2]         <int>: 0
[ssh:Windows 10] [2] 
[ssh:Windows 10] [2]     C:/Users/Admin/amqp_ci_rcv_odo-windows-e2e-pr-build_PR_4658/repo/tests/helper/helper_run.go:34
[ssh:Windows 10] [2] ------------------------------
[ssh:Windows 10] [2] Created dir: C:\Users\Admin\849480791
...
[ssh:Windows 10] [2] [odo] 
 X  Waiting for component to start [4m]
[ssh:Windows 10] [2] [odo]  X  waited 4m0s but couldn't find running pod matching selector: 'deploymentconfig=sb-jar-test-app'
```
Issue 2.  https://github.com/openshift/odo/issues/3256
```
[ssh:Windows 10] [2] odo devfile supported tests
[ssh:Windows 10] [2] C:/Users/Admin/amqp_ci_rcv_odo-windows-e2e-pr-build_PR_4658/repo/tests/e2escenarios/e2e_devfile_test.go:24
[ssh:Windows 10] [2]   odo debug support for devfile components
[ssh:Windows 10] [2]   C:/Users/Admin/amqp_ci_rcv_odo-windows-e2e-pr-build_PR_4658/repo/tests/e2escenarios/e2e_devfile_test.go:66
[ssh:Windows 10] [2]     Verify output debug information for java-openliberty debug works [It]
[ssh:Windows 10] [2]     C:/Users/Admin/amqp_ci_rcv_odo-windows-e2e-pr-build_PR_4658/repo/tests/e2escenarios/e2e_devfile_test.go:73
[ssh:Windows 10] [2] 
[ssh:Windows 10] [2]     No future change is possible.  Bailing out early after 300.461s.
[ssh:Windows 10] [2]     Running odo.exe with args [odo push --context C:\Users\Admin\692823498/projectDir]
[ssh:Windows 10] [2]     Expected
[ssh:Windows 10] [2]         <int>: 1
[ssh:Windows 10] [2]     to match exit code:
[ssh:Windows 10] [2]         <int>: 0
[ssh:Windows 10] [2] 
[ssh:Windows 10] [2]     C:/Users/Admin/amqp_ci_rcv_odo-windows-e2e-pr-build_PR_4658/repo/tests/helper/helper_run.go:34
[ssh:Windows 10] [2] ------------------------------
[ssh:Windows 10] [2] Created dir: C:\Users\Admin\692823498
[ssh:Windows 10] [2] Setting KUBECONFIG=C:\Users\Admin\692823498\config
[ssh:Windows 10] [2] Creating a new project: e2e-devfile-test73brp
[ssh:Windows 10] [2] Running odo.exe with args [odo project create e2e-devfile-test73brp -w -v4]

[ssh:Windows 10] [2] [odo] I0426 09:17:28.957046   11160 deployments.go:152] Waiting for deployment "tpqovd" rollout to finish: 0 of 1 updated replicas are available...
[ssh:Windows 10] [2] [odo] I0426 09:17:28.957046   11160 deployments.go:159] Waiting for deployment spec update to be observed...
[ssh:Windows 10] [2] [odo] 
 X  Failed to start component with name "tpqovd". Error: Failed to create the component: error while waiting for deployment rollout: timeout while waiting for tpqovd deployment roll out
[ssh:Windows 10] [2] [odo]  X  Waiting for component to start [5m]
```

**Comment by prietyc123** [comment]
@anandrkskd Thanks for the pr. AFAIU this pr will partially resolve #4412 and #4408 . However As per the pr subject this pr will close the issue #4412 and #4408 . I think if you want to mention a part of its fix you may write for example Fixes - part of #4412 in pr template.

**Comment by prietyc123** [comment]
```INFO[2021-04-30T11:23:13Z] [ssh:Fedora 32]     /home/fedora/amqp_ci_rcv_odo-minikube-pr-build_PR_4658/repo/tests/integration/devfile/cmd_devfile_storage_test.go:174 
INFO[2021-04-30T11:23:13Z] [ssh:Fedora 32]                              
INFO[2021-04-30T11:23:13Z] [ssh:Fedora 32]     Unexpected error:   
INFO[2021-04-30T11:23:13Z] [ssh:Fedora 32]         <*exec.Error | 0xc0000ffe20>: { 
INFO[2021-04-30T11:23:13Z] [ssh:Fedora 32]             Name: "oc",      
INFO[2021-04-30T11:23:13Z]                                              
INFO[2021-04-30T11:23:13Z] [ssh:Fedora 32]             Err: {           
INFO[2021-04-30T11:23:13Z] [ssh:Fedora 32]                 s: "executable file not found in $PATH", 
INFO[2021-04-30T11:23:13Z] [ssh:Fedora 32]             },               
INFO[2021-04-30T11:23:13Z] [ssh:Fedora 32]         }                    
INFO[2021-04-30T11:23:13Z] [ssh:Fedora 32]         exec: "oc": executable file not found in $PATH 
INFO[2021-04-30T11:23:13Z] [ssh:Fedora 32]     occurred             
INFO[2021-04-30T11:23:13Z] [ssh:Fedora 32]                              
INFO[2021-04-30T11:23:13Z] [ssh:Fedora 32]     /home/fedora/amqp_ci_rcv_odo-minikube-pr-build_PR_4658/repo/tests/helper/helper_run.go:27 
```
This has been fixed in https://github.com/openshift/odo/pull/4677

/test psi-kubernetes-integration-e2e

**Comment by girishramnani** [comment]
is this still WIP @anandrkskd ?

File: scripts/run_script_e2e.sh [hunk 1]
```diff
@@ -0,0 +1,92 @@
...
+set +x
+oc login -u developer -p ${OCP4X_DEVELOPER_PASSWORD} --insecure-skip-tls-verify ${OCP4X_API_URL}
+set -x
+
+# # Integration tests
+#shout "Running integration Tests"
+make test-operator-hub || error=true
+#remove comments from test-integration & test-integration-devfile once #4606 https://github.com/openshift/odo/issues/4606 is resolved 
+#make test-integration || error=true
+#make test-integration-devfile || error=true
```
[the remark this answers is not in the document]
Comment (anandrkskd) [reply scripts/run_script_e2e.sh:62]: we are facing issues in which `odo push` is failing, this is mostly failing on integration,integration-devfile tests. so for now we are just testing those parts that doesn't require `odo push` in  integration, integration-devfile suites.

**Comment by dharmit** [comment]
@anandrkskd how relevant is this going to be w.r.t IBM Cloud? Would the changes proposed in this PR be useful for it or would we have to do things differently through a different PR?

**Comment by rnapoles-rh** [comment]
> @anandrkskd how relevant is this going to be w.r.t IBM Cloud? Would the changes proposed in this PR be useful for it or would we have to do things differently through a different PR?

@dharmit  When we switch to use the IBM Cloud cluster we just change the target server and use the API key to login. I just found out that the API token for the Cluster running in the IBM Cloud is expiring. Checking on that. But this does not affect this PR that is using the cluster running in PSI.

**Comment by anandrkskd** [comment]
Splitting the PR into two parts 
- Script for automation (https://github.com/openshift/odo/pull/4877)
- Fix tests related to platform(https://github.com/openshift/odo/pull/4873)
Thus, closing this issue.
/close

**Pull request closed without merging**